### DISCUSSION TITLE
Open a record page from clicking a relation manager record

### DISCUSSION THREAD

**Discussion by RobQuistNL** [filamentphp/filament#13872] [Help] [answered]
### Package

Table builder

### Package Version

v3.2.62

### How can we help you?

I have a RelationManager that shows relational records within an entity.

I want to open the view page of that linked entity by clicking the URL. The magic `recordAction` method asks for a closure or a string. I've tried URL's, classname strings, returning URL's in functions, static ViewAction::class things and whatnot - all throw errors or don't work.

I just want a click to open that page. How do I do that?

```php
class SystemCommandsRelationManager extends RelationManager
{
    protected static string $relationship = 'systemCommands';

    protected static ?string $recordTitleAttribute = 'id';

    public function table(Table $table): Table
    {
        return SystemCommandResource::table($table)
            ->poll('1s')
            ->recordAction(/* something goes here but no idea what */);
    }
}
```

**Comment by leandrocfe** [comment, marked as answer]
```php
->recordUrl(fn ($record) => YourResource::getUrl('view', ['record' => $record->id]))
```

**Reply by RobQuistNL** [reply]
Thank you!
